analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ConditionEvaluatesToConstantBase.cs

```diff
@@ -67,11 +67,15 @@ public override ProgramState ConditionEvaluated(SymbolicContext context)
         return context.State;
     }
 
-    private bool IsNotIgnored(ProgramState state, IOperation operation) =>
-        operation.Kind is not OperationKindEx.Literal
-        && operation.TrackedSymbol(state) is not IFieldSymbol { IsConst: true }
-        && !IsDiscardPattern(operation)
-        && !operation.Syntax.Ancestors().Any(x => IsInsideUsingDeclaration(x) || IsLockStatement(x));
+    private bool IsIgnored(ProgramState state, IOperation operation) =>
+        operation.Kind is OperationKindEx.Literal
+        || IsVarPattern(operation)
+        || IsDiscardPattern(operation)
+        || operation.TrackedSymbol(state) is IFieldSymbol { IsConst: true }
+        || operation.Syntax.Ancestors().Any(x => IsInsideUsingDeclaration(x) || IsLockStatement(x));
+
+    private static bool IsVarPattern(IOperation operation) =>
+        operation.AsIsPattern()?.Pattern.WrappedOperation.AsDeclarationPattern()?.MatchesNull is true;
```

Comment:
```suggestion
        operation.AsIsPattern()?.Pattern.WrappedOperation?.AsDeclarationPattern()?.MatchesNull is true;
```

Reply:
Do you have a test case where `WrappedOperation` would be `null`?